src/ansiblelint/_prerun.py

```diff
@@ -102,14 +102,16 @@ def _prepare_ansible_paths() -> None:
 
     if os.path.exists("plugins/modules") and "plugins/modules" not in library_paths:
         library_paths.append("plugins/modules")
-
-    if os.path.exists(".cache/collections"):
+    if (
+        os.path.exists(".cache/collections")
+        and ".cache/collections" not in collection_list
```

Comment:
Maybe there should be a helper function `add_if_exists_and_not_present` (happy bikeshedding ;) ) which receives the list and the new path.